plugins/uefi-capsule/fu-uefi-bootmgr.c

```diff
@@ -292,6 +293,165 @@ fu_uefi_copy_asset (const gchar *source, const gchar *target, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_uefi_bootmgr_setup_grub (FuDevice *device,
+			    const gchar *esp_path,
+			    const gchar *target_app,
+			    GError **error)
+{
+	g_autofree gchar *grub_reboot = NULL;
+	g_autofree gchar *grub_target = NULL;
+	g_autofree gchar *sysconfdir = NULL;
+
+	/* verify GRUB snippet exists */
+	sysconfdir = fu_common_get_path (FU_PATH_KIND_SYSCONFDIR);
+	grub_target = g_build_filename (sysconfdir, "grub.d", "35_fwupd", NULL);
+	if (!g_file_test (grub_target, G_FILE_TEST_EXISTS)) {
+		g_autofree gchar *buf = NULL;
+		g_autofree gchar *datadir = NULL;
+		g_autofree gchar *grub_source = NULL;
+		g_autofree gchar *mount = NULL;
+		g_autofree gchar *update_grub = NULL;
+		g_autoptr(GString) str = NULL;
+
+		/* read config snippet */
+		datadir = fu_common_get_path (FU_PATH_KIND_DATADIR_PKG);
+		grub_source = g_build_filename (datadir, "fwupd.grub.conf", NULL);
+		if (!g_file_get_contents (grub_source, &buf, NULL, error))
+			return FALSE;
+
+		/* replace ESP path with what we detected */
+		str = g_string_new (buf);
+		fu_common_string_replace (str, "##EFIPATH##", target_app);
+		fu_common_string_replace (str, esp_path, "");
+		fu_common_string_replace (str, "##ESP##", esp_path);
+		if (!g_file_set_contents (grub_target, str->str, -1, error))
+			return FALSE;
+		if (chmod (grub_target, 0755) == -1) {
+			g_set_error (error,
+				     G_IO_ERROR,
+#ifdef HAVE_ERRNO_H
+				     g_io_error_from_errno (errno),
+#else
+				     G_IO_ERROR_FAILED,
+#endif
+				     "chmod failed for %s", grub_target);
+			return FALSE;
+		}
+
+		/* refresh GRUB configuration */
+//Fedora needs
+// grub2-mkconfig -o "$(readlink -e /etc/grub2.conf)"
+//IIUC
```

Comment:
@martinezjavier is this true?

Reply:
It is true. But I wonder if Fedora's grub2 package should just ship `update-grub`, which is just a stub for this command.

Reply:
At least in Ubuntu it doesn't use `/etc/grub2.conf` though I think - it just goes right to something like `/boot/grub/grub.conf`

Reply:
So maybe the right answer is a heuristic to check a list of paths to see if they exist and just use `grub2-mkconfig` with them?

Reply:
Just checked on Ubuntu they do put it in `/boot/grub/grub.cfg` directly.
```
$ cat /usr/sbin/update-grub
#!/bin/sh
set -e
exec grub-mkconfig -o /boot/grub/grub.cfg "$@"
```

Reply:
Yes, I think we should do the same in Fedora and just update the GRUB config directly instead of the symlink in `/etc`. But I still wonder if is worth to also ship an update-grub wrapper script... @vathpela what do you think ?

Reply:
Oh it's a cross filesystem symlink on Fedora?  In that case perhaps we should just do `grub-mkconfig -o /boot/grub/grub.cfg` across the board to make this as compatible as possible across distros (including old releases).

Reply:
Can you re-check what I've pushed to the branch now?  Hopefully this works on Debian/Ubuntu/Fedora all.